src/libraries/System.Text.Json/src/System/Text/Json/Serialization/BidirectionalDictionary.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace System.Text.Json
+{
+    internal sealed class BidirectionalDictionary<TKey, TValue>
```

Comment:
I was just trying to consolidate the logic and also was following James' approach.
https://github.com/JamesNK/Newtonsoft.Json/blob/master/Src/Newtonsoft.Json/Utilities/BidirectionalDictionary.cs

But using only one dictionary I dont think so, wouldn't be better to have two dictionaries to avoid some boxing?
```cs
private Dictionary<string, object> _referenceKeyValueMapper;
private Dictionary<object, string> _referenceValueKeyMapper;
```

Reply:
Actually, the reason why I did all of this was to rely in a `ReferenceResolver` that we may want to expose in a not-so-distant future (probably in the new converter model). You could replace the BidirectionalDictionary for two individual dictionaries living in the `DefaultReferenceResolver`. I guess I can do that.

Reply:
Or you can use one single dictionary, but then we would have boxing, if that is ok, then we can do store all on a `Dictionary<object, object>`.